### PR TITLE
Use dynamic client everywhere

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

Changes the machine-controller to use controller-runtime everywhere, except in the node evictions, as evictions are a subresource the dynamic client cant handle.

The goal of this is to simplify the machine-controller, working towards #545 .

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

Sorry for the size of the PR :grimacing: 

/assign @mrIncompetent 

```release-note
none
```

### CODE REVIEW

**Comment by alvaroaleman** [comment]
/hold

We should do at least one further machine-controller release before we merge this

File: cmd/controller/main.go [hunk 1]
```diff
@@ -501,14 +488,12 @@ func startControllerViaLeaderElection(runOptions controllerRunOptions) error {
...
-			runOptions.nodeLister,
-			runOptions.machineInformer,
-			runOptions.machineLister,
-			runOptions.secretSystemNsLister,
+			mgr.GetClient(),
+			mgr.GetRecorder("machine-controller"),
 			runOptions.metrics,
 			runOptions.prometheusRegisterer,
+			runOptions.machineInformer,
+			runOptions.nodeInformer,
```
Comment (mrIncompetent) [inline cmd/controller/main.go]: Why do we still need those informers?

File: pkg/controller/machine/machine.go [hunk 2]
```diff
@@ -84,12 +82,9 @@ const (
 
 // Controller is the controller implementation for machine resources
 type Controller struct {
-	kubeClient    kubernetes.Interface
-	machineClient clusterv1alpha1clientset.Interface
-
-	nodesLister          listerscorev1.NodeLister
-	machinesLister       clusterlistersv1alpha1.MachineLister
-	secretSystemNsLister listerscorev1.SecretLister
+	ctx        context.Context
```
Comment (mrIncompetent) [inline pkg/controller/machine/machine.go]: What is this context for?

File: cmd/controller/main.go (hunk 1 above)
Comment (alvaroaleman) [reply cmd/controller/main.go]: For triggering the reconciliation. Currently, this controller is not a controller-runtime controller, the only thing the PR changes is that the dynamic client is used everywhere (except on the eviction where that is not possible).

I was thinking this makes the reviewing easier, but I can also extend this PR to make it a controller-runtime controller. WDYT?

File: pkg/controller/machine/machine.go (hunk 2 above)
Comment (alvaroaleman) [reply pkg/controller/machine/machine.go]: Its used in the api calls and will get canceled when the leader lock was lost

File: cmd/controller/main.go (hunk 1 above)
Comment (mrIncompetent) [reply cmd/controller/main.go]: I would prefer keeping the PR size as is :grimacing:

**Pull request merged**